### DISCUSSION TITLE
Mocking Api

### DISCUSSION THREAD

**Discussion by ddrobins** [mozts2005/ZendeskApi_v2#569] [General]
I am setting up unit tests for some services which involve this API. I'd like to be able to mock the calls, but I am getting an error on dependency injection that IZendeskAPI doesn't have an accessible constructor. Do you have any recommendations for how to handle this? Or could it be an enhancement request?
Thanks!

**Comment by mozts2005** [comment]
Did You find a fix for this?

**Comment by ddrobins** [comment]
I have not. It seems like an issue is my class is getting instantiated with the constructor with parameters instead of the parameter less one. So when I run tests locally it’s fine but once the code goes live I get the Zendesk error.
